### PR TITLE
docs: Backport 2.0.0 upgrade notes to 24.7 release

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/623

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: docs/modules/nifi/partials/supported-versions.adoc [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // This is a separate file, since it is used by both the direct NiFi-Operator documentation, and the overarching
 // Stackable Platform documentation.
 
-- 2.0.0-M4 (experimental)
+- 2.0.0-M4 (experimental) - Please note that you need to upgrade to at least 1.27.x before upgrading to 2.0.x!
```
Comment (maltesander) [inline docs/modules/nifi/partials/supported-versions.adoc:5]: I think 1.25 is the magic barrier for 2.0.0 prep.
```suggestion
- 2.0.0-M4 (experimental) - Please note that you need to upgrade to at least 1.25.x before upgrading to 2.0.x!
```

Comment (sbernauer) [reply docs/modules/nifi/partials/supported-versions.adoc:5]: https://cwiki.apache.org/confluence/display/NIFI/Migration+Guidance
Actually it's a mess:
1.24 for 2.0.0-M1
1.25 for 2.0.0-M2
1.26 for 2.0.0-M3
1.26 for 2.0.0-M4

WDYT of playing safe and stating 1.27 here?

Comment (maltesander) [reply docs/modules/nifi/partials/supported-versions.adoc:5]: Ah then my knowledge came from the M2... yeah then ignore this and lets play it safe :)

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by fhennig** [comment]
PR **666** :open_mouth: :smiling_imp:

**Comment by sbernauer** [comment]
:smiling_imp:

**Pull request merged**